.github/Dockerfile.base

```diff
@@ -0,0 +1,44 @@
+FROM ubuntu:22.04
+SHELL ["/bin/bash", "-c"]
+
+# Set environment variables
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Install dependencies
+RUN apt-get update && apt-get install -y \
+    software-properties-common \
+    build-essential \
+    python3-dev \
+    python3-venv \
+    python3-pip \
+    git \
+    git-lfs \
+    libhwloc-dev \
+    pandoc \
+    libtbb-dev \
+    libcapstone-dev \
+    pkg-config \
+    linux-tools-generic \
+    ninja-build \
+    wget \
+    cmake \
```

Comment:
Do we need cmake here as you're already installing it through pip below?

Reply:
I'm not sure, but I think installing just "apt cmake" didn't work as it may have installed two different CMake versions. The following section is copied and pasted from the tt-mlir project because at the time, I didn't have the dependencies for tt-forge-fe. We should review this section and consider building the image from the tt-mlir base image instead of using Ubuntu. This way, we can simply add tt-forge specific dependencies or move to a common compiler base image.
Definitly something to improve in future.